server.go

```diff
@@ -258,7 +258,10 @@ func (s *Server) handleNextGame(rw http.ResponseWriter, req *http.Request) {
 			gh = newHandle(newGame(request.GameID, randomState(words)), s.Store)
 			s.games[request.GameID] = gh
 		} else if request.CreateNew {
-			nextState := nextGameState(gh.g.GameState)
+			var nextState GameState
+			gh.update(func(g *Game) {
+				nextState = nextGameState(g.GameState)
+			})
```

Comment:
In this case, we're replacing the GameHandle in the `s.games` map, not just updating the existing game handle. I think we _do_ want to wake up waiting `/game-state` goroutines like you do here, but ideally we want it to be _after_ we've installed the new handle, and we want the other goroutines to reacquire the handle so they notice the new `GameHandle`.

I'm going to think on this one 🤔